edb/server/protocol/args_ser.pyx

```diff
@@ -156,14 +157,26 @@ cdef WriteBuffer recode_bind_args(
 
             frb_read(&in_buf, 4)  # reserved
             in_len = hton.unpack_int32(frb_read(&in_buf, 4))
-            out_buf.write_int32(in_len)
-
             if in_len < 0:
                 # This means argument value is NULL
                 if param.required:
                     raise errors.QueryError(
                         f"parameter ${param.name} is required")
 
+            # If the param has encoded tuples, we need to decode them
+            # and reencode them as arrays of scalars.
+            if param.sub_params:
+                # HACK: Rewind the frb, since _decode_tuple_args needs the
+                # length present.
+                frb_read(&in_buf, -4)
```

Comment:
Ugh, let's add `frb_seek` instead.

Reply:
I think I might just try to adjust things to not need it